awx/main/models/inventory.py

```diff
@@ -820,9 +820,47 @@ def _get_related_jobs(self):
 
 
 class HostMetric(models.Model):
-    hostname = models.CharField(primary_key=True, max_length=512)
+    hostname = models.CharField(unique=True, max_length=512)
     first_automation = models.DateTimeField(auto_now_add=True, null=False, db_index=True, help_text=_('When the host was first automated against'))
     last_automation = models.DateTimeField(db_index=True, help_text=_('When the host was last automated against'))
+    last_deleted = models.DateTimeField(null=True, db_index=True, help_text=_('When the host was last deleted'))
+    automated_counter = models.IntegerField(default=0, help_text=_('How many times was the host automated'))
+    deleted_counter = models.IntegerField(default=0, help_text=_('How many times was the host deleted'))
+    deleted = models.BooleanField(
+        default=False, help_text=_('Boolean flag saying whether the host is deleted and therefore not counted into the subscription consumption')
+    )
+    used_in_inventories = models.IntegerField(null=True, help_text=_('How many inventories contain this host'))
+
+    objects = models.Manager()
+    active_objects = HostMetricActiveManager()
+
+    def get_absolute_url(self, request=None):
+        return reverse('api:host_metric_detail', kwargs={'pk': self.pk}, request=request)
+
+    def soft_delete(self):
+        if not self.deleted:
+            self.deleted_counter = (self.deleted_counter or 0) + 1
+            self.last_deleted = now()
+            self.deleted = True
+            self.save()
```

Comment:
I do really strongly prefer to use `update_fields`, since it will avoid any conflicts with other updates going on. In this case, I don't think any other process even changes these deletion-related fields, which is perfect. Requests run in a transaction. It'll be great to never have this edit a value that the task system also edits.

Reply:
I have no problem to change it to `update_fields`. 
I don't understand the rest of your comment, if it's just comment, or suggestion :)

Reply:
@AlanCoding  Interesting, I'd guess update_fields and .save() must be equvivalent? No 2 processes can touch the same in-memory structure of this (unless we have some badly done multithreading there), so in the end both will just generate a SQL query and send it. (with last update winning)